api/utils/singlepoint_helper.py

```diff
@@ -0,0 +1,78 @@
+"""Helper functions for performing singeploint calculations."""
+
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from typing import Any
+
+from janus_core.calculations.single_point import SinglePoint
+from janus_core.helpers.janus_types import Architectures, Properties
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+def convert_ndarray_to_list(obj: dict | list | np.ndarray | float) -> dict:
```

Comment:
Two minor things

- All the `dict`s could probably have a bit more detail e.g. `[str, Any]` (or more specific if possible, but I'm not sure you can here)
- I'd maybe go with something other than `obj` - I don't think it's strictly wrong, but `obj` makes it sounds like something to do with a class. Maybe just `data`?